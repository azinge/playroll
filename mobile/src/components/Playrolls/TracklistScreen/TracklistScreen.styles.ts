import { StyleSheet, Platform } from 'react-native';
import { ThemeConsumer } from 'react-native-elements';

export default StyleSheet.create({
  screenContainer: {
    flex: 1,
    backgroundColor: '#fff',
  },

  // Header
  headerView: {
    alignItems: 'center',
    marginBottom: 10,
  },
  titleView: {
    marginTop: Platform.OS == "ios" ? 20 : 4,
    width: '100%',
  },
  innerTitleView: {
    alignItems: 'center',
  },
  titleText: {
    alignItems: 'center',
    color: '#FFF',
    fontSize: 20,
    fontWeight: 'bold',
    margin: 10,
  },
  genPlaylistButton: {
    width: '90%',
  },

  // Scroll View Content
  rollCardContainer: {
    marginTop: 3,
    marginBottom: 10,
  },
  rollCardTitle: {
    fontSize: 16,
    // marginTop: 0,
  },
  trackView: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    margin: 2,
  },
  trackImage: {
    height: 25,
    width: 25,
    marginRight: 5,
  },

  // Footer
  footerButton: {
    
  },
  footerView: {
    backgroundColor: '#fff',
    flex: 1,
    position: 'absolute',
    bottom: 0, // stick to bottom
    left: 0,   // stretch to left
    right: 0,  // stretch to right
    height: 60,
    alignItems: 'center',
    justifyContent: 'center',
    zIndex: 1, // float on top
  }
});
