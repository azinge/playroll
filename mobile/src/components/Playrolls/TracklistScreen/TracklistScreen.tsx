/**
 * Application component for Playroll mobile application.
 */

import React from 'react';
import { Text, View, Image, ScrollView, SafeAreaView } from 'react-native';
import { Card, Button } from 'react-native-elements';
import { NavigationScreenProp } from 'react-navigation';

import {
  GeneratePlaylistMutation,
  GetTracklistQuery,
} from '../../../graphql/requests/Tracklist';

import styles from './TracklistScreen.styles';

export interface Props {
  navigation?: NavigationScreenProp<{}>;
}

interface State {}

// TODO: re-style this screen
export default class TracklistScreen extends React.Component<Props, State> {
  render() {
    const nav = this.props.navigation;
    const tracklistID =
      nav && nav.state && nav.state.params && nav.state.params.tracklistID;
    const playlistName =
      nav && nav.state && nav.state.params && nav.state.params.playrollName;
    return (
      <SafeAreaView style={styles.screenContainer}>
        <GetTracklistQuery variables={{ id: tracklistID }}>
          {({ loading, error, data }) => {
            return (
              <View style={{ flex: 1, marginTop: 20 }}>
                <View style={styles.headerView}>
                  {/* TODO: improve title styling */}
                  <Text style={styles.titleText}>{`${playlistName} > Tracklist`}</Text>
                  <GeneratePlaylistMutation
                    variables={{
                      tracklistID,
                      playlistName,
                    }}
                  >
                    {(generatePlaylist, { data }) => (
                      <Button
                        title='Generate Playlist'
                        containerStyle={styles.genPlaylistButton}
                        onPress={() => {
                          generatePlaylist();
                        }}
                      />
                    )}
                  </GeneratePlaylistMutation>
                </View>
                <ScrollView>
                  {data &&
                    data.tracklist &&
                    data.tracklist.compiledRolls &&
                    data.tracklist.compiledRolls.map(compiledRoll => {
                      return (
                        compiledRoll &&
                        compiledRoll.data &&
                        compiledRoll.data.tracks &&
                        compiledRoll.data.tracks.length > 0 && (
                          <Card
                            key={compiledRoll.id}
                            title={`Compiled roll #${compiledRoll.id}`}
                            titleStyle={styles.rollCardTitle}
                          >
                            {compiledRoll.data.tracks.map(track => {
                              return (
                                track && (
                                  <View style={styles.trackView} key={track.providerID}>
                                    <Image
                                      style={styles.trackImage}
                                      source={{ uri: track.cover }}
                                    />
                                    <Text>{track.name}</Text>
                                    {/* <Text>{track.type}</Text> */}
                                  </View>
                                )
                              );
                            })}
                          </Card>
                        )
                      );
                    })}
                </ScrollView>
              </View>
            );
          }}
        </GetTracklistQuery>
      </SafeAreaView>
    );
  }
}
