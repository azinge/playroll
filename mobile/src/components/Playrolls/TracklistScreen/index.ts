import TracklistScreen from "./TracklistScreen";
export default TracklistScreen;
