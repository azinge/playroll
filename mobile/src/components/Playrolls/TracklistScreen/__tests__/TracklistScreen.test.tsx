import * as React from "react";
import renderer from "react-test-renderer";

import TracklistScreen from "../TracklistScreen";

it("renders correctly", () => {
  const tree = renderer.create(<TracklistScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
