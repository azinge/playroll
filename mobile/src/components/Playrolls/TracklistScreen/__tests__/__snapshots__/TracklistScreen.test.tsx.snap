// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginTop": 20,
      }
    }
  >
    <View>
      <Text>
        undefined - Tracklist
      </Text>
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 20,
                    },
                    undefined,
                    Object {
                      "fontSize": 17.5,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Generate Playlist
            </Text>
          </View>
        </View>
      </View>
    </View>
    <RCTScrollView>
      <View />
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
